torch/csrc/jit/profiling_record.cpp

```diff
@@ -18,7 +18,73 @@ ProfileOp* ProfilingRecord::createProfileNode(
   return pn;
 }
 
-void ProfilingRecord::instrumentBlock(Block* block) {
+static void unprofileGraphInputs(const std::shared_ptr<Graph> &graph) {
+  for (auto i : graph->inputs()) {
+    if (i->type()->isSubtypeOf(TensorType::get())) {
+      i->setType(unshapedType(i->type()));
+    }
+  }
+}
+
+static void unprofileBlock(Block* start_block) {
+  std::vector<Block*> stack;
+  stack.push_back(start_block);
+
+  while (!stack.empty()) {
+    Block* block = stack.back();
+    stack.pop_back();
+
+    for (auto n : block->nodes()) {
+      for (auto o : n->outputs()) {
+        if (o->type()->isSubtypeOf(TensorType::get())) {
+          o->setType(unshapedType(o->type()));
+        }
+      }
+      for (auto ib : n->blocks()) {
```

Comment:
If you want to make this less verbose, you could do `blocks.insert(blocks.end(), n->blocks().begin(), n->blocks().end());`

Personally I'm fine with at as-is.

Reply:
ah, that's a good catch.